src/Sparrow.Server/Platform/Posix/CGroupHelper.cs

```diff
@@ -0,0 +1,315 @@
+                continue;
+
+            return match.Groups["path"].Value;
+        }
+
+        return null;
+    }
+}
+
+public class CGroupV2 : CGroup
+{
+    protected override string MemoryLimitFileName => "memory.max";
+    protected override string MemoryUsageFileName => "memory.current";
+    protected override string MaxMemoryUsageFileName => "memory.peak";
+
+    protected override string FindCGroupPathForMemory() => FindCGroupPath();
+    
+    private static string FindCGroupPath()
+    {
+        FindHierarchyMount(out var mountRoot, out var mountPath);
+        var pathForSubsystem = FindCGroupPathForSubsystem();
+        
+        return CombinePaths(mountRoot, mountPath, pathForSubsystem);
+    }
+    private static void FindHierarchyMount(out string mountRoot, out string mountPath)
+    {
+        foreach (var match in FindHierarchyMount())
+        {
+            mountRoot = match.Groups["mountroot"].Value;
+            mountPath = match.Groups["mountpath"].Value;
+            return;
+        }
+
+        throw new CGroupException($"Couldn't find hierarchy mount in {PROC_MOUNTINFO_FILENAME}");
+    }
+    
+    // 0::/user.slice/user-1000.slice/user@1000.service/apps.slice/apps-org.gnome.Terminal.slice/vte-spawn-d7794050-ce4a-451b-92c2-a2433019409e.scope
+    private static readonly Regex FindCGroupPathForSubsystemReg = new Regex(@"^\d+::(?<path>.*)$", RegexOptions.Compiled);
+    private static string FindCGroupPathForSubsystem()
+    {
+        foreach (var line in File.ReadLines(PROC_CGROUP_FILENAME))
+        {
+            var match = FindCGroupPathForSubsystemReg.Match(line);
+            if(match.Success == false)
+                continue;
+                
+            return match.Groups["path"].Value;
+        }
+
+        return null;
+    }
+}
+
+public class UnidentifiedCGroup : CGroup
+{
+    private readonly string _errorMessage;
+    private DateTime _lastLog = DateTime.MinValue;
+    protected override string MemoryLimitFileName => null;
+    protected override string MemoryUsageFileName => null;
+    protected override string MaxMemoryUsageFileName => null;
+
+    public UnidentifiedCGroup(string errorMessage)
+    {
+        _errorMessage = errorMessage;
+    }
+
+    protected override string FindCGroupPathForMemory()
+    {
+        if (_lastLog + TimeSpan.FromMinutes(10) < DateTime.UtcNow)
+        {
+            _lastLog = DateTime.UtcNow;
+            if(Logger.IsOperationsEnabled)
+                Logger.Operations(_errorMessage);
+        }
+
+        return null;
+    }
+}
+
+public static class CGroupHelper
+{
+    private static readonly Logger Logger = LoggingSource.Instance.GetLogger("Server", nameof(CGroupHelper));
+    public static readonly CGroup CGroup = GetCGroup();
+
+    private static CGroup GetCGroup()
+    {
+        const uint TMPFS_MAGIC = 0x01021994;
+        const uint CGROUP2_SUPER_MAGIC = 0x63677270;
```

Comment:
Magic 🧙‍♂️

Reply:
I just used the same name
https://github.com/omajid/dotnet-runtime/blob/392dd8b01aab6b3731fe7e5fbacdfd218c57da78/src/coreclr/src/gc/unix/cgroup.cpp#L41

Reply:
Yeah, all good :)